components/datadog/agentparams/params.go

```diff
@@ -197,11 +197,33 @@ func WithTelemetry() func(*Params) error {
 func WithFakeintake(fakeintake *fakeintake.ConnectionExporter) func(*Params) error {
 	return func(p *Params) error {
 		// configure metrics and check run intake
+		args := make([]interface{}, 12) // number of %s in the configuration (horribly hardcoded)
+		for idx := range args {
+			args[idx] = fakeintake.Host
+		}
 		extraConfig := pulumi.Sprintf(`dd_url: http://%s:80
 logs_config.logs_dd_url: %s:80
 logs_config.logs_no_ssl: true
 logs_config.force_use_http: true
-process_config.process_dd_url: http://%s:80`, fakeintake.Host, fakeintake.Host, fakeintake.Host)
+process_config.process_dd_url: http://%s:80
+database_monitoring.metrics.logs_dd_url: %s:80
+database_monitoring.metrics.logs_no_ssl: true
+database_monitoring.activity.logs_dd_url: %s:80
+database_monitoring.activity.logs_no_ssl: true
+database_monitoring.samples.logs_dd_url: %s:80
+database_monitoring.samples.logs_no_ssl: true
+network_devices.metadata.logs_dd_url: %s:80
+network_devices.metadata.logs_no_ssl: true
+network_devices.snmp_traps.forwarder.logs_dd_url: %s:80
+network_devices.snmp_traps.forwarder.logs_no_ssl: true
+network_devices.netflow.forwarder.logs_dd_url: %s:80
+network_devices.netflow.forwarder.logs_no_ssl: true
+container_lifecycle.logs_dd_url: %s:80
+container_lifecycle.logs_no_ssl: true
+container_image.logs_dd_url: %s:80
+container_image.logs_no_ssl: true
+sbom.logs_dd_url: %s:80
+sbom.logs_no_ssl: true`, args...)
```

Comment:
`fmt.Sprinf` can use indexed arguments which would make this much more readable, and avoid the hacky arg slice at the beginning (https://go.dev/play/p/7xVTDDBUTV9)

Reply:
One day a wise person said `tous les héros ne portent pas de cape` and now I understand what he meant, thanks ❤️